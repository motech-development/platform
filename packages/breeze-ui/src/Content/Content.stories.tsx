import { withA11y } from '@storybook/addon-a11y';
import { storiesOf } from '@storybook/react';
import BaseStyles from '../BaseStyles/BaseStyles';
import Typography from '../Typography/Typography';
import Content from './Content';

const stories = storiesOf('Content', module);

stories.addDecorator(withA11y);

stories.add('Basic content', () => (
  <>
    <BaseStyles />

    <Content>
      <Typography component="h1" variant="h1">
        Hello world
      </Typography>
    </Content>
  </>
));
