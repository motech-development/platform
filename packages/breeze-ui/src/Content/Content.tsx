import { FC, memo, ReactNode } from 'react';
import styled from 'styled-components';

const Wrapper = styled.section`
  background-color: #f8f8f8;
  color: #000;
  display: block;
  padding: 1rem;
`;

export interface IContentProps {
  children: ReactNode;
}

const Content: FC<IContentProps> = ({ children }) => (
  <Wrapper>{children}</Wrapper>
);

export default memo(Content);
