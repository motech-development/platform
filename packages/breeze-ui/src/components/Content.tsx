import { ElementType, forwardRef } from 'react';
import { Box, PolymorphicComponentProps } from 'react-polymorphic-box';
import { classNames } from '../utils/className';

export type TContentProps<E extends ElementType> = PolymorphicComponentProps<
  E,
  {}
>;

const defaultElement = 'div';

const Content = forwardRef(
  <E extends ElementType = typeof defaultElement>(
    { className = '', ...rest }: TContentProps<E>,
    ref: typeof rest.ref,
  ) => (
    <Box
      as={defaultElement}
      className={classNames(
        'max-w-7xl mx-auto pb-12 px-4 sm:px-6 lg:px-8',
        className,
      )}
      ref={ref}
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...rest}
    />
  ),
);

export default Content;
