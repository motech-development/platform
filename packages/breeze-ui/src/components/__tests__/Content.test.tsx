import { render } from '@testing-library/react';
import Content from '../Content';

describe('Content', () => {
  it('should render correctly as a div by default', () => {
    const { container } = render(
      <Content>
        <h1>Hello world</h1>
      </Content>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly as another element', () => {
    const { container } = render(
      <Content as="section" className="test-class">
        <h1>Hello world</h1>
      </Content>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
