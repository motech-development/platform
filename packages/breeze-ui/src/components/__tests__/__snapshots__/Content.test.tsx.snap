// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render correctly as a div by default 1`] = `
<div
  class="max-w-7xl mx-auto pb-12 px-4 sm:px-6 lg:px-8"
>
  <h1>
    Hello world
  </h1>
</div>
`;

exports[`Content should render correctly as another element 1`] = `
<section
  class="max-w-7xl mx-auto pb-12 px-4 sm:px-6 lg:px-8 test-class"
>
  <h1>
    Hello world
  </h1>
</section>
`;
